presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationPushDown.java

```diff
@@ -132,51 +139,95 @@ public void testOffsetPushDown() throws ExecutionException, InterruptedException
         ResultWithQueryId<MaterializedResult> queryResult = queryRunner.executeWithQueryId(getSession(), sql);
-        assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), 998);
+        assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
+                .describedAs("with dump:\n%s", dumpResultAndData(topicNameCreateTime, recordMessage, queryRunner, sql, queryResult))
+                .isEqualTo(998);
 
         // timestamp_upper_bound_force_push_down_enabled set as true.
         Session sessionWithUpperBoundPushDownEnabled = Session.builder(getSession())
                 .setSystemProperty("kafka.timestamp_upper_bound_force_push_down_enabled", "true")
                 .build();
 
         queryResult = queryRunner.executeWithQueryId(sessionWithUpperBoundPushDownEnabled, sql);
-        assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), 2);
+        assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
+                .describedAs("with dump:\n%s", dumpResultAndData(topicNameCreateTime, recordMessage, queryRunner, sql, queryResult))
+                .isEqualTo(2);
     }
 
-    @Test
+    @Test(retryAnalyzer = RetryAnalyzerForTimestampTest.class)
     public void testTimestampLogAppendModePushDown() throws ExecutionException, InterruptedException
     {
-        Pair<String, String> timePair = createTimestampTestMessages(topicNameLogAppend);
+        RecordMessage recordMessage = createTimestampTestMessages(topicNameLogAppend);
         DistributedQueryRunner queryRunner = (DistributedQueryRunner) getQueryRunner();
         // ">= startTime" insure including index 2, "< endTime"  insure excluding index 4;
         String sql = String.format("SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s' and _timestamp < timestamp '%s'",
-                topicNameLogAppend, timePair.first(), timePair.second());
+                topicNameLogAppend, recordMessage.getStartTime(), recordMessage.getEndTime());
         ResultWithQueryId<MaterializedResult> queryResult = queryRunner.executeWithQueryId(getSession(), sql);
-        assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), 2);
+
+        assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
+                .describedAs("with dump:\n%s", dumpResultAndData(topicNameLogAppend, recordMessage, queryRunner, sql, queryResult))
+                .isEqualTo(2);
+    }
+
+    private String dumpResultAndData(String topic, RecordMessage recordMessage, DistributedQueryRunner queryRunner, String sql, ResultWithQueryId<MaterializedResult> queryResult)
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("dump 5 expected messages:").append("\n");
```

Comment:
Drop `dump` prefix from dump section headers.
It seems `5` is also not needed here.

Also I think it would be more readable if capture the result int `List<String>` and use `Joiner.on("/n").join(....)` at the end.
Some vertical whitespace would make this method a bit readable too.